src/util/Collection.js

```diff
@@ -357,9 +360,14 @@ class Collection extends Map {
     if (!collection) return false;
     if (this === collection) return true;
     if (this.size !== collection.size) return false;
-    return !this.find((value, key) => {
+    return this.every((value, key) => {
+      if (!collection.has(key)) return false;
       const testVal = collection.get(key);
-      return testVal !== value || (testVal === undefined && !collection.has(key));
+      try {
```

Comment:
Objects/arrays should not be tested deeply - simply checking for basic reference equality was fine.